controller/internal/controller/consumer_controller.go

```diff
@@ -181,3 +197,73 @@ func (r *ConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	return controller.Complete(r)
 }
+
+// checkConsumerConflicts Perform full conflict detection and rebuild the index
+func (r *ConsumerReconciler) checkConsumerConflicts(ctx context.Context) error {
+	r.keyIndex.mu.Lock()
+	defer r.keyIndex.mu.Unlock()
+
+	// Clear old indexes
+	r.keyIndex.index = make(map[string]map[string]map[string]string)
+
+	// Get all Consumer
+	var allConsumers mosniov1.ConsumerList
+	if err := r.ResourceManager.List(ctx, &allConsumers); err != nil {
+		return fmt.Errorf("list consumers failed: %w", err)
+	}
+
+	// Perforated detection
+	for _, consumer := range allConsumers.Items {
+		if !consumer.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		if err := r.indexConsumer(consumer); err != nil {
+			return fmt.Errorf("consumer %s/%s: %w", consumer.Namespace, consumer.Name, err)
+		}
+	}
+	return nil
+}
+
+// indexConsumer Index a single Consumer and detect conflicts
+func (r *ConsumerReconciler) indexConsumer(consumer mosniov1.Consumer) error {
+	for pluginName, plugin := range consumer.Spec.Auth {
+		key, err := extractPluginKey(plugin.Config.Raw)
+		if err != nil {
+			return fmt.Errorf("invalid plugin %s config: %w", pluginName, err)
+		}
+
+		// Inertly initialize the three-level map
+		if r.keyIndex.index[consumer.Namespace] == nil {
+			r.keyIndex.index[consumer.Namespace] = make(map[string]map[string]string)
+		}
+		if r.keyIndex.index[consumer.Namespace][pluginName] == nil {
+			r.keyIndex.index[consumer.Namespace][pluginName] = make(map[string]string)
+		}
+
+		// collision detect
+		if existing, exists := r.keyIndex.index[consumer.Namespace][pluginName][key]; exists {
+			return fmt.Errorf(
+				"key conflict: plugin=%s key=%s (already used by %s)",
+				pluginName, key, existing,
+			)
+		}
+		r.keyIndex.index[consumer.Namespace][pluginName][key] = consumer.Name
+	}
+	return nil
+}
+
+func extractPluginKey(raw []byte) (string, error) {
```

Comment:
Better to find a way to call the plugin's Index method like https://github.com/mosn/htnn/blob/c8e8eeb91b8612f33bb3c53b552622ff19995c7e/api/internal/consumer/consumer_manager.go#L90C12-L90C23, instead of hardcoding the field names.